libs/application/core/src/lib/ApplicationTemplateHelper.ts

```diff
@@ -58,8 +70,22 @@ export class ApplicationTemplateHelper<
    * @param event A state machine event
    * returns [hasChanged, newState, newApplication] where newApplication has the updated state value
    */
-  changeState(event: Event<TEvents>): [boolean, string, Application] {
-    this.initializeStateMachine()
+  changeState(
+    event: Event<TEvents>,
+    emailService: any,
+  ): [boolean, string, Application] {
+    this.initializeStateMachine(undefined, {
+      services: {
+        emailService: async (
+          context: TContext,
+          event: TEvents,
+          { src }: InvokeMeta,
+        ) => {
+          const template = src.template(context)
+          return emailService.sendEmail(template)
```

Comment:
maybe we have to NOT do this if there is no emailTemplate?

Reply:
This function is not called unless the
`invoke: {
          src: {
            type: 'emailService',
            emailTemplate: assignParentTemplate,
          },
        },
`
part is added to the state, meaning it will invoke the emailService when transitioning to this state, but adding a check for the emailTemplate is a good thing, I'm also going to fix the types.